validation-test/compiler_crashers_2/0083-rdar31163470-2.swift

```diff
@@ -0,0 +1,24 @@
+// RUN: not --crash %target-swift-frontend -primary-file %s -emit-ir
+// REQUIRES: asserts
+
+protocol C {
+  associatedtype I
+}
+
+protocol PST {
+  associatedtype LT : C
+}
+
+protocol SL {
+  associatedtype S : PST
+}
+
+struct PEN<_S : PST> : SL {
+  typealias S = _S
+  let l: _S.LT.I
+}
+
+struct PE<N : SL> {
+  let n: N
+  static func c<S>(_: PE<N>) where N == PEN<S> {}
```

Comment:
I take it back... `S: PST` should not be in the minimized signature, because it's derivable from the concrete conformance of `N: SL` where `N == PEN<S>`. Per @slavapestov , this is an issue where we're doing a `subst` on a type that isn't canonical w.r.t. the signature.